### PR TITLE
python3Packages.nix-prefetch-github: 6.0.1 -> 7.0.0

### PR DESCRIPTION
**Pull request by tjni**
###### Description of changes

https://github.com/seppeljordan/nix-prefetch-github#v700

This was started by @seppeljordan in https://github.com/NixOS/nixpkgs/pull/242092. I've taken it over to handle merge conflicts and get it across the finish line.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 243018` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>solana-validator</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>element-desktop</li>
    <li>element-desktop-wayland</li>
    <li>emacsPackages.evil-textobj-tree-sitter</li>
    <li>emacsPackages.evil-tree-edit</li>
    <li>emacsPackages.helm-tree-sitter</li>
    <li>emacsPackages.pygn-mode</li>
    <li>emacsPackages.symex</li>
    <li>emacsPackages.tree-sitter</li>
    <li>emacsPackages.tree-sitter-ess-r</li>
    <li>emacsPackages.tree-sitter-indent</li>
    <li>emacsPackages.tree-sitter-ispell</li>
    <li>emacsPackages.tree-sitter-langs</li>
    <li>emacsPackages.tsc</li>
    <li>lemmy-server</li>
    <li>nix-prefetch-github (python310Packages.nix-prefetch-github)</li>
    <li>nix-prefetch-github.dist (python310Packages.nix-prefetch-github.dist)</li>
    <li>python311Packages.nix-prefetch-github</li>
    <li>python311Packages.nix-prefetch-github.dist</li>
    <li>schildichat-desktop</li>
    <li>schildichat-desktop-wayland</li>
  </ul>
</details>

File: pkgs/applications/blockchains/solana-validator/update.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env nix-shell
-#! nix-shell -i oil -p jq sd nix-prefetch-github ripgrep
+#! nix-shell -i oil -p jq moreutils nix-prefetch-github sd
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/solana-validator/update.sh:2]: we should use gnused instead

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/blockchains/solana-validator/update.sh (hunk 1 above)
Comment (tjni) [reply pkgs/applications/blockchains/solana-validator/update.sh]: good idea, done!

**Comment by tjni** [comment]
@adisbladis If you have time, can you review the two lemmy commits that touch the update script?

**Comment by adisbladis** [comment]
> @adisbladis If you have time, can you review the two lemmy commits that touch the update script?

Those parts LGTM.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**